assets/js/html5-details-polyfill.js

```diff
@@ -0,0 +1,174 @@
+var ua = window.navigator.userAgent;
+var isIE = /MSIE|Trident/.test(ua);
+
+
+if (isIE) {
```

Comment:
it's better to do this via feature detection than browser tests - e.g. edge seems to be considering support for this in the future.

Looks like the feature detection here is super heavy so I'm happy to only do feature-detection on the browsers that _might not_ support it, as long as it only polyfills if it really isn't supported.

Let's just add a comment to this effect when we add edge support.

Reply:
Thanks @jbothma 

You're right that feature detection is better than browser tests, but I've never found a good way to achieve this in practice - it's almost always been better to do a browser test. That being said, if you have a suggestion for how to achieve it I'm more than happy to try it out.

I've come across this, but never used it: https://modernizr.com/
It looks pretty good, let me know what you think.